### ISSUE TITLE
Modify tbl_df subclass print in a package

### ISSUE THREAD

**Issue by urswilke** [r-lib/pillar#650]
Not sure if this topic belongs here and rather a comment than an issue. I was struggling to have a customized print of a tibble subclass `"my_subclass"` in a package using pillar.

If I understand correctly, the solution was to add these roxygen2 comments (and then `devtools::document()`):

``` r
#' @importFrom
#' @export
```
for instance
``` r
#' @importFrom pillar tbl_sum
#' @export
tbl_sum.my_subclass <- function(x, ...) {
  c("custom_title_string")
}
```

I was just wondering if it might be worth to add such a comment to the docs, because I lost too much time with this. For further explanation you can also check out [this dummy package](https://github.com/urswilke/testr6print) that I made to test this and which helped me to find the solution (the R6 part there isn't relevant for this topic I think).

**Comment by krlmlr** [comment]
Good point. Are you suggesting tweaking https://pillar.r-lib.org/articles/extending.html, or other texts?

**Comment by urswilke** [comment]
Yes I think that would be the best place. It was the article I read most when learning this.

**Comment by moodymudskipper** [comment]
To create in a package a custom S3 method for a tibble subclass we need to import the generic, and export the method. This is not specific to pillar but is a detail that is easy to overlook and can confuse users.

We wish to improve the documentation to avoid this confusion.
